### PR TITLE
CONN-6606 update release reference for x12 v2.5.0

### PR DESCRIPTION
**Pull request by rongjiwang**
doc ref for x12 2.5.0 . 
REVIEW AND MERGE - ORDER 1

### CODE REVIEW

**Review by vijayang — DISMISSED** [review]

File: x12-edi/2.5/antora.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: x12-edi-connector
-version: '2.1'
-display_version: 2.1 (Mule 4)
+version: '2.5'
```
Comment (kbostrom) [inline x12-edi/2.5/antora.yml:2]: Wow, did the version jump from .1 to .5?

File: x12-edi/2.5/antora.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+display_version: 2.5 (Mule 4)
 title: X12 EDI Connector
```
Comment (kbostrom) [inline x12-edi/2.5/antora.yml:4]: Follow the template for title format.
I can't remember, but if the template uses the full connector name here, please do so.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -2,7 +2,7 @@
 :page-aliases: connectors::x12-edi/x12-edi-connector.adoc
 
 Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Premium] +
-X12 EDI Connector Version 2.1.1
+X12 EDI Connector Version 2.5.0
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc:5]: ditto prev. two questions/comments

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -294,6 +295,10 @@ Writer options let you specify the separator characters used when writing messag
 * Whether to use supplied values for control segment information (control numbers, in particular). The default behavior is to always generate control numbers when writing, this allows you to instead use values you have chosen.
 * Whether to use the default output MIME type `application/plain`, or the X12-specific alternative `application/edi-x12`.
 * Whether to enforce length limits on input data. The default behavior is to do length validations.
+* Whether to enforce conditional rules on input data. The default behavior is to ignore conditional rules.
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: GLOBAL format fix:
```suggestion
* Whether to enforce conditional rules on input data +
The default behavior is to ignore conditional rules.
```

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -294,6 +295,10 @@ Writer options let you specify the separator characters used when writing messag
...
+* The key value pair of interchange control key and interchange control number.
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]:
```suggestion
* The key-value pair of interchange control key and interchange control number
```

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -294,6 +295,10 @@ Writer options let you specify the separator characters used when writing messag
...
+* The key value pair of group control key and group control number.
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]:
```suggestion
* The key-value pair of group control key and group control number
```

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 7]
```diff
@@ -294,6 +295,10 @@ Writer options let you specify the separator characters used when writing messag
...
+* The key value pair of transaction control key and transaction control number.
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]:
```suggestion
* The key-value pair of transaction control key and transaction control number
```

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 8]
```diff
@@ -406,6 +411,8 @@ Parser parameters control the parser operation and the types of error conditions
...
 |reportSegmentErrors="true" |Report segment error details to sender in 997/999 flag. If true, the details of any segment errors are included in a generated 997/999. If false, the details are not included.
 |includeFASchema="true" |Expect 997 or 999 Functional Acknowledgments and include 997/999 schema. By default, the schema for 997/999 acknowledgment transaction sets is automatically included in the set of schemas used by the X12 EDI Module. If false, you need to directly specify the 997 and/or 999 schema if you want these to be processed as input. The schemas used for generating 997 or 999 Functional Acknowledgments are hard coded and cannot be modified.
+|acknowledgmentSchemaPath="" |Expect the path either from the file system or the classpath. If the path is specified, it overwrites the default acknowledgment.
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: It overrides the default path?
How does a path override an acknowledgment?

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 9]
```diff
@@ -71,6 +71,7 @@ Enforce minimum and maximum lengths for Writer values. If true, a transaction wi
 | Initial transaction number a| Number |  The initial Transaction Set Control Number used for outgoing messages. |  1 |
 | Request acknowledgments a| Boolean |  Request acknowledgments i.e. TA1 for sent transactions flag (ISA14). If true, TA1 acknowledgments are requested for all sent transactions. |  false |
 | Default usage indicator a| String |  Default interchange usage indicator (ISA15): I for Information, P for Production Data, and T for Test Data. |  P |
+| Enforce conditional rules a| Boolean | Enforce conditional rules are applied in a transaction. If true, a transaction with values not meeting conditional rules is rejected. If false, the values are used anyway and the transaction is not rejected. | false |
```
Comment (kbostrom) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-reference.adoc]:
```suggestion
| Enforce conditional rules a| Boolean | Conditional rules are applied in a transaction. If `true`, a transaction with values not meeting conditional rules is rejected. If `false`, the noncompliant values are used anyway and the transaction is not rejected. | false |
```

**Review by kbostrom — DISMISSED** [review]
Edited

File: x12-edi/2.5/antora.yml (hunk 1 above)
Comment (mcherem) [reply x12-edi/2.5/antora.yml:2]: I"ts a long story.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 3 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc:5]: Sqme

File: x12-edi/2.5/antora.yml (hunk 2 above)
Comment (mcherem) [reply x12-edi/2.5/antora.yml:4]: This is in the yaml file, so I don't want to change it.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 4 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: fixed

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 5 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 7 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 8 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: path

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-reference.adoc (hunk 9 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-reference.adoc]: fixed

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 10]
```diff
@@ -292,13 +292,19 @@ Writer options let you specify the separator characters used when writing messag
 * The starting transaction set control number (used only when no current value is found in the object store).
 * The default interchange usage indicator (ISA15), used when no value is present in the message data.
 * Whether to request interchange acknowledgments (ISA14).
-* Whether to use supplied values for control segment information (control numbers, in particular). The default behavior is to always generate control numbers when writing, this allows you to instead use values you have chosen.
+* Whether to use supplied values for control segment information (control numbers, in particular).
+
+The default behavior is to always generate control numbers when writing, this allows you to instead use values you have chosen.
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc:297]: Starting here the bullet lists are not properly formatted and won't render correctly. Need to fix the bullet lists.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 11]
```diff
@@ -292,13 +292,19 @@ Writer options let you specify the separator characters used when writing messag
...
-* The key value pair of interchange control key and interchange control number.
-* The key value pair of group control key and group control number.
-* The key value pair of transaction control key and transaction control number.
+* Whether to enforce length limits on input data.
+
+The default behavior is to do length validations.
+* Whether to enforce conditional rules on input data.
+
+The default behavior is to ignore conditional rules.
+* The key-value pair of interchange control key and interchange control number.
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: No periods on sentence fragments (see Kim's edit)

**Review by vijayang — APPROVED** [review]

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**